org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CallHierarchyHandler.java

```diff
@@ -269,6 +270,24 @@ private List<CallHierarchyOutgoingCall> getOutgoingCallItemsAt(String uri, int l
 		return result;
 	}
 
+	/*
+	 * Wrapper around MethodWrapper to handle the case where it is called more than once.
+	 * As subsequent calls are fast, MethodWrapper.getCalls(..) does not report any progress, which
+	 * seems like an upstream bug.
+	 */
+	private static MethodWrapper[] getCallsAndHandleProgress(MethodWrapper wrapper, IProgressMonitor monitor) {
+		String oldValue = monitor.toString();
+		// MethodWrapper does not report progress when called on a cached value
+		// This can result in failing to report progress completion when a set of calls are cached
+		MethodWrapper[] calls = wrapper.getCalls(monitor);
```

Comment:
why not call `monitor.done()` directly here? It should do no harm to invoke monitor.done() two times.

Reply:
Worth a shot. Definitely simpler.

Reply:
It works, and doesn't emit an extra notification. :partying_face: